### ISSUE TITLE
Supported Hierarchical Departments

### ISSUE THREAD

**Issue by jaymzh** [rubocop/rubocop#10373]
## Is your feature request related to a problem? Please describe.

It would be very useful for Rubocop to be able to supported hierarchical/nested departments.

For example, if I have a department MyExtension/Foo and MyExtension/Bar it would be useful to be able to enable/disable MyExtension as a whole,

In particular this comes up with things like [Cookstyle](https://github.com/chef/cookstyle/) which is a tool built on top of rubocop to do Chef correctness linting. They have many departments all of the form Chef/<whatever>. It would be useful to be able to turn all of those on or off.

In my particular case since for standard ruby-correctness, I tend to stick much closer to Rubocop defaults than Cookstyle defaults, but Cookstyle includes a ton of its own defaults for rubocop rules, for my cookstyle runs, I actually want to disable all rules, then enable all Chef/* rules, ala something like this:

```yaml
AllCops:
  DisabledByDefault: true

Chef:
  Enabled: true
```

Unfortunately today I have to list every single department manually and if they add one, I have to add it to my config.

## Additional context

See further discussion of this in https://github.com/rubocop/rubocop/issues/9752

**Comment by jaymzh** [comment]
CC @jonas054  who wanted a tag in this.
